File: C/PRAK404-2510817210005-Daffa-Dziibaan-Khoiri.c
#include <stdio.h>
#include <stdlib.h>
int main(){


    int pilihan;
    float nilai_1;
    float nilai_2;
    float hasil;
    char *operasi;
    do{
        printf("\nPilih Program\n1. Penjumlahan\n2. Pengurangan\n3. Perkalian\n4. Pembagian\n5. Exit\n\n");
        printf("Masukkan Pilihan : ");
        scanf("%d", &pilihan);

        if(pilihan == 5){
        printf("Terima kasih sudah menggunakan kalkulator DAFFA DZIIBA'AN KHOIRI");
            break;
        }
        if(pilihan < 1 || pilihan > 5){
            printf("Input anda salah, silahkan coba lagi");
            continue;
        }
        printf("Masukkan nilai pertama : ");
        scanf("%f", &nilai_1);
        
        do{
            printf("Masukkan nilai kedua : ");
            scanf("%f", &nilai_2);

            if(pilihan == 4 && nilai_2 == 0){
                printf("Tidak bisa membagi angka dengan 0 silahkan input angka ulang\n\n");
                continue;
            }
        }
        while (nilai_2 == 0);
        
        

        switch (pilihan)
        {
        case 1:
            hasil = nilai_1 + nilai_2;
            operasi = "Penjumlahan";    
            break;
        case 2:
            hasil = nilai_1 - nilai_2;
            operasi = "Pengurangan";
            break;
        case 3:
            hasil = nilai_1 * nilai_2;
            operasi = "Perkalian";
            break;
        case 4:
            hasil = nilai_1 / nilai_2;
            operasi = "Pembagian";
            break;
        default:
            continue;
        }

        printf("Hasil %s antara %.2f dan %.2f adalah %.2f\n", operasi, nilai_1, nilai_2, hasil);
        continue;
    }
    while (pilihan != 5);
    
    return 0;
}

File: Python/PRAK404-2510817210005-Daffa-Dziibaan-Khoiri.py
while True :
    print(
"""

Pilih pilihan
1. Penjumlahan
2. Pengurangan
3. Perkalian
4. Pembagian
5. Exit

""")
    pilihan = int(input("Masukkan Pilihan : "))

    if pilihan == 5 :
       print('Terima kasih sudah menggunakan kalkulator DAFFA DZIIBA"AN KHOIRI')
       break

    if pilihan < 1 or pilihan > 5 :
        print("Input anda salah, silahkan coba lagi")
        continue

    nilai_pertama = int(input("Masukkan nilai pertama : "))
    

    while True :
        nilai_kedua = int(input("Masukkan nilai pertama : "))
        if pilihan == 3 and nilai_kedua == 0 :
            print("TIdak bisa membagi angka dengan 0 silahkan input angka ulang\n")
            continue
        else :
            break
    
    match pilihan :
        case 1 :
            hasil = nilai_pertama + nilai_kedua
            operasi = "Penjumlahan"
            
        case 2 :
            hasil = nilai_pertama - nilai_kedua
            operasi = "Pengurangan"
            
        case 3 :
            hasil = nilai_pertama * nilai_kedua
            operasi = "Perkalian"
            
        case 4 :
            hasil = nilai_pertama / nilai_kedua
            operasi = "Pembagian"
            
        case _:
            break


    print(f"Hasil {operasi} antara {nilai_pertama:.2f} dan {nilai_kedua:.2f} adalah {hasil:.2f}\n")
    continue

